ti/instrumentation/instrumentation.go

```diff
@@ -84,6 +84,14 @@ func getTestSelection(ctx context.Context, runner TestRunner, config *api.RunTes
 	} else {
 		log.Infoln(fmt.Sprintf("Running tests selected by Test Intelligence: %s", selection.Tests))
 	}
+
+	if tiConfig.GetParseSavings() {
+		if config.RunOnlySelectedTests {
+			tiConfig.WriteSavingsState(stepID, ti.TI, ti.OPTIMIZED)
+		} else {
+			tiConfig.WriteSavingsState(stepID, ti.TI, ti.FULL_RUN)
```

Comment:
This is the case where you haven't set any optimization features (disabled TI) so it should be ti.DISABLED. FULL_RUN is when config.RunOnlySelectedTests is set (user enabled TI) but we ran all tests (selected == total). Let me know if I got this wrong.

Reply:
We would only hit this code if user asked for TI. Ref: https://github.com/harness/lite-engine/blob/main/ti/instrumentation/instrumentation.go#L219

`config.RunOnlySelectedTests` would be `false` here in case we get an error with TI. You can check the above lines.

Reply:
Added comments to clear the confusion

Reply:
Ah okay. The config.RunOnlySelectedTests is confusing.